src/client/replication_ddl_client.cpp

```diff
@@ -81,18 +82,51 @@ namespace replication {
 
 using tp_output_format = ::dsn::utils::table_printer::output_format;
 
-replication_ddl_client::replication_ddl_client(const std::vector<dsn::rpc_address> &meta_servers)
+replication_ddl_client::replication_ddl_client(const std::vector<dsn::host_port> &meta_servers)
+    : _dns_resolver(new dns_resolver())
 {
     _meta_server.assign_group("meta-servers");
     for (const auto &m : meta_servers) {
-        if (!_meta_server.group_address()->add(m)) {
+        if (!_meta_server.group_host_port()->add(m)) {
             LOG_WARNING("duplicate adress {}", m);
         }
     }
 }
 
 replication_ddl_client::~replication_ddl_client() { _tracker.cancel_outstanding_tasks(); }
 
+void replication_ddl_client::set_meta_servers_leader()
```

Comment:
We should selecting the leader host actively. All of `call()` we should transfer `host_port` to `rpc_address`，but not pointer passing should not change group leader by `on_recv_reply()`.

Reply:
`dsn_rpc_call(dns_resolver->resolve_address(hp), rpc_call);`